File: AdminServiceDotNET/Models/AuditLogs.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AdminServiceDotNET.Models
{
    [Table("audit_logs")]
    public class AuditLogs
    {
        [Key]
        [Column("log_id")]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Required]
        public long Id { get; set; }

        [Required]
        [Column("service_name")]
        [MaxLength(50)]
        public ServiceName ServiceName { get; set; }

        [Required]
        [Column("action_name")]
        [MaxLength(50)]
        public  AuditAction Action { get; set; }

        
        [Column("user_email")]
        [Required]
        [MaxLength(150)]
        public string UserEmail { get; set; }

        [Required]
        [Column("user_role")]
        [MaxLength(50)]
        public UserRole Role { get; set; }

        [Column("details")]
        public string Details { get; set; }

        [Required]
        [Column("created_at")]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    }
}

File: oep_spring_backend/src/main/java/com/oep/entities/AuditLogs.java
package com.oep.entities;

import jakarta.persistence.AttributeOverride;
import jakarta.persistence.AttributeOverrides;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

/*log_id INT PRIMARY KEY AUTO_INCREMENT,
service_name VARCHAR(50),
user_email VARCHAR(150),
action VARCHAR(150),
details JSON,
created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP*/
@Entity
@Table(name = "audit_logs")

@AttributeOverrides({@AttributeOverride(name = "createdOn", column = @Column(name = "created_at")),
		@AttributeOverride(name = "id", column = @Column(name = "log_id"))})
@Getter
@Setter
public class AuditLogs extends BaseEntity {

    @Enumerated(EnumType.STRING)
    @Column(name = "service_name", nullable = false)
    private ServiceName serviceName;

    @Column(name = "user_email", length = 150, nullable = false)
    private String userEmail;
    
    @Column(name = "user_role", length = 50, nullable = false)
    @Enumerated(EnumType.STRING)
    private UserRole role;
    
    @Enumerated(EnumType.STRING)
    @Column(name = "action", nullable = false)
    private AuditAction action;
    
    @Column(name = "details", columnDefinition = "json")
    private String details;
}
